src/orchestrator.ts

```diff
@@ -22,19 +22,25 @@ export class Orchestrator {
     // As a result, we are currently constrained to initialize all of our data in the `handle` method.
     constructor(public fn: (context: IOrchestrationFunctionContext) => IterableIterator<unknown>) {}
 
-    public listen(): (context: IOrchestrationFunctionContext) => Promise<OrchestratorState> {
+    public listen(): (
+        context: IOrchestrationFunctionContext,
+        orchestrationTrigger: DurableOrchestrationBindingInfo
+    ) => Promise<OrchestratorState> {
         return this.handle.bind(this);
     }
 
-    private async handle(context: IOrchestrationFunctionContext): Promise<OrchestratorState> {
+    private async handle(
+        context: IOrchestrationFunctionContext,
+        orchestrationTrigger: DurableOrchestrationBindingInfo
+    ): Promise<OrchestratorState> {
         this.taskOrchestrationExecutor = new TaskOrchestrationExecutor();
         const orchestrationBinding = Utils.getInstancesOf<DurableOrchestrationBindingInfo>(
-            context.bindings,
+            { trigger: orchestrationTrigger },
             new DurableOrchestrationBindingInfoReqFields() as DurableOrchestrationBindingInfo
         )[0];
 

```

Comment:
I'm confused about how `getInstancesOf` is working after this change. 
My understanding is that `getInstancesOf` used to search for any values of its generic type (in this case that's `DurableOrchestrationBindingInfo` inside of `context`.

However, now it seems like we're getting the binding info as a parameter itself, so do we still need to call `getInstancesOf`?

Reply:
You're right that the main function of `getInstancesOf` was to search for the value in a collection. However, it also had validation of types, and, crucially, conversion of timestamps: https://github.com/Azure/azure-functions-durable-js/blob/7320b2d57ceaa53fbc0ac4212fcca21141acbfb9/src/utils.ts#L22. I figured I don't wanna mess with this right now, and just wanted to make the fewest changes necessary to get to a working POC. I do agree there could be a lot of refactoring of this function now that we're not searching for a value anymore. I could do that in this PR if you feel strongly about it.

Reply:
Makes sense, good call. Thanks!